processor/metricstransformprocessor/metrics_transform_processor_otlp.go

```diff
@@ -540,4 +572,7 @@ func transformMetric(metric pmetric.Metric, transform internalTransform) {
 			}
 		}
 	}
+
+	// Consider metric invalid if all its data points were removed after applying the operations.
+	return isMetricEmpty || countDataPoints(metric) > 0
```

Comment:
Can we true true earlier if `isMetricEmpty` is true?  Seems more performant to exit early if we know we can.

Reply:
Actually we still can perform some operations on an empty, e.g. renaming

Reply:
Looks like no other operations other than renaming can do anything on empty metrics, so we can potentially move it up right after the renaming, but I'm a bit worried that, if another operation is added that can do something on empty metrics, we will skip it. Also empty/invalid metrics are pretty rare case, so I wouldn't optimize code for them.

Reply:
Ok lets leave it as is.